src/test/java/dog/pawbook/testutil/EditProgramDescriptorBuilder.java

```diff
@@ -0,0 +1,69 @@
+package dog.pawbook.testutil;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import dog.pawbook.logic.commands.EditProgramCommand.EditProgramDescriptor;
+import dog.pawbook.model.managedentity.Name;
+import dog.pawbook.model.managedentity.program.Program;
+import dog.pawbook.model.managedentity.program.Session;
+import dog.pawbook.model.managedentity.tag.Tag;
+
+/**
+ * A utility class to help with building EditProgramDescriptor objects.
+ */
+public class EditProgramDescriptorBuilder {
```

Comment:
It will be good to adhere to the DRY principle and apply some OOP like in the main codebase. That way setName and setTags can all be simplified. It helps that EditEntityDescriptor is already in place so there isn't as much tweaking to do. This was also one of the reason I rewrote the Edit command implementation to ensure minimal repetition, avoiding potential points of failure when copying blindly and not keeping necessary parts updated.